enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,701 @@
+
+As an alternative, configuring BGP peers and which route-targets to import would eliminate the need to manually
+configure routes in the host, and would allow dynamic routing updates based on changes in the provider’s network.
+
+### Exporting Routes into the Provider Network
+
+There exists a need for provider networks to learn routes directly to services and pods today in Kubernetes. More
+specifically, MetalLB is already one solution where load balancer IPs are advertised by BGP to provider networks. The
+goal of this RFE is to not duplicate or replace the function of MetalLB. MetalLB should be able to interoperate with
+OVN-Kubernetes, and be responsible for advertising services to a provider’s network.
+
+However, there is an alternative need to advertise pod IPs on the provider network. One use case is integration with 3rd
+party load balancers, where they terminate a load balancer and then send packets directly to OCP nodes with the
+destination IP address being the pod IP itself. Today these load balancers rely on custom operators to detect which node
+a pod is scheduled to and then add routes into its load balancer to send the packet to the right node.
+
+By integrating BGP and advertising the pod subnets/addresses directly on the provider network, load balancers and other
+entities on the network would be able to reach the pod IPs directly.
+
+### Datapath Performance
+
+In cases where throughput is a priority, using the underlay directly can eliminate the need for tunnel encapsulation,
+and thus reducing the overhead and byte size of each packet. This allows for greater throughput.
+
+### Multi-homing, Link Redundancy, Fast Convergence
+
+BGP can use multi-homing with ECMP routing in order to provide layer 3 failover. When a link goes down, BGP can reroute
+via a different path. This functionality can be coupled with BFD in order to provide fast failover.
+
+### User Stories
+
+ * As a user I want to be able to leverage my existing BGP network to dynamically learn routes to pods in my Kubernetes
+   cluster.
+ * As a user, rather than having to maintain routes with NM State manually in each Kubernetes node, as well as being
+   constrained to using local gateway mode for respecting user defined routes; I want to use BGP so that I can dynamically
+   advertise egress routes for the Kubernetes pod traffic in either gateway mode.
+ * As a user where maximum throughput is a priority, I want to reduce packet overhead by not having to encapsulate
+   traffic with Geneve.
+ * As a baremetal or egress IP user, I do not want to have to restrict my nodes to the same layer 2 segment and prefer
+   to use a pure routing implementation to handle advertising virtual IP (VIP) movement across nodes.
+
+### Goals
+
+* To provide a user facing API to allow configuration of iBGP or eBGP peers, along typical BGP configuration including
+  communities, route filtering, etc.
+* Support for advertising Egress IP addresses.
```

Comment:
Advertising egress IP addresses to who? What functionality does this enable?

Reply:
this is useful when an egressip-ed pod sends packet to an endpoint within the fabric: in order to allow the return traffic to reach the pod, the endpoint's reply needs to find its way to the node.
Since egressip is dynamic, here we announce it via bgp so the fabric will learn how to drive the return traffic to the proper node.

Reply:
in the user stories we have:
* As a baremetal or egress IP user, I do not want to have to restrict my nodes to the same layer 2 segment and prefer
   to use a pure routing implementation to handle advertising virtual IP (VIP) movement across nodes.

which I think covers it...although we removed support for baremetal VIP from this enhancement so I'll clean that up a bit